base/src/main/java/org/aya/resolve/context/PhysicalModuleContext.java

```diff
@@ -43,6 +39,17 @@ public PhysicalModuleContext(@NotNull Context parent, @NotNull ImmutableSeq<Stri
     this.moduleName = moduleName;
   }
 
+  @Override
+  public @Nullable ContextUnit getUnqualifiedLocalMaybe(
+    @NotNull String name,
+    Stmt.@Nullable Accessibility accessibility,
+    @NotNull SourcePos sourcePos
+  ) {
+    var permitted = permits.getOption(name);
+    if (permitted.isDefined()) return permitted.get();
+    return ModuleContext.super.getUnqualifiedLocalMaybe(name, accessibility, sourcePos);
```

Comment:
Why we need this `permits`?

Reply:
I guess the `permits` is for the code below?
```
open data N | zero | suc N
open data N | zero | suc N
```

Reply:
It is for the code:
```
private data N | zero | suc N
```

Reply:
emmm

Reply:
That's because the context is checking the accessibility without knowing the being-resolved definition. Right?

Reply:
Because `data N` is private. However, I removed the accessibility checking.

Reply:
I still think checking acc is useful. If only there's a simple way to achieve that....

Reply:
The way I impl it is too dirty, I have no better idea.

Reply:
Okay. We can save it for later